xarray/core/groupby.py

```diff
@@ -554,11 +589,7 @@ def factorize(self, group) -> T_FactorizeOut:
         full_index = binned.categories
         uniques = np.sort(pd.unique(binned_codes))
         unique_values = full_index[uniques[uniques != -1]]
-        group_indices = [
-            g for g in _codes_to_groups(binned_codes, len(full_index)) if g
-        ]
-
-        if len(group_indices) == 0:
+        if (binned_codes == -1).all():
```

Comment:
This check is redundant since it is already present a few lines up.
Remove it or revert back to `len(group_indices) == 0`?

Reply:
Ah yes, bad rebase.

I kept it for the more specialized error message we can raise in the binning case.